### PR TITLE
reject tx with too large gas limit

### PR DESCRIPTION
**Pull request by yihuang**
## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/ethereum/backend/backend.go [hunk 1]
```diff
@@ -607,6 +607,12 @@ func (e *EVMBackend) SendTransaction(args types.SendTxArgs) (common.Hash, error)
 
 	msg := args.ToTransaction()
 
+	// FIXME hotfix, reject tx with too high gas limit
```
Comment (yihuang) [inline rpc/ethereum/backend/backend.go:610]:
```suggestion
	// hotfix, reject tx with too high gas limit
	// FIXME remove after real fix is used: https://github.com/tharsis/ethermint/pull/751
```

File: rpc/ethereum/namespaces/eth/api.go [hunk 2]
```diff
@@ -374,6 +374,12 @@ func (e *PublicAPI) SendRawTransaction(data hexutil.Bytes) (common.Hash, error)
 		return common.Hash{}, fmt.Errorf("invalid transaction type %T", tx)
 	}
 
+	// FIXME hotfix, reject tx with too high gas limit
```
Comment (yihuang) [inline rpc/ethereum/namespaces/eth/api.go:377]:
```suggestion
	// hotfix, reject tx with too high gas limit
	// FIXME remove after real fix is used: https://github.com/tharsis/ethermint/pull/751
```

**Review by tomtau — COMMENTED** [review]
where is that `10000000` gas from? is it the max block gas network parameter? is it possible to fetch the network parameter directly instead?

**Comment by yihuang** [comment]
> where is that `10000000` gas from? is it the max block gas network parameter? is it possible to fetch the network parameter directly instead?

It's the current max block gas parameter, it's discussed before that this value should be enough for all the known use cases.
The network parameter could be lifted later when the tx execution optimization is applied though.

**Review by tomtau — COMMENTED** [review]
ok for the hotfix, just one good thing to change is not to have an undocumented magic number in the fix -- instead if the block max gas can be obtained directly from the context (e.`<something>`) or have a named constant that has a comment with a link to https://github.com/crypto-org-chain/cronos-mainnet/blob/master/cronosmainnet_25-1/genesis.json#L8

**Comment by yihuang** [comment]
> ok for the hotfix, just one good thing to change is not to have an undocumented magic number in the fix -- instead if the block max gas can be obtained directly from the context (e.`<something>`) or have a named constant that has a comment with a link to https://github.com/crypto-org-chain/cronos-mainnet/blob/master/cronosmainnet_25-1/genesis.json#L8

Made a constant for it, and added more comments.

**Review by tomtau — APPROVED** [review]

**Pull request merged**